universal-login-wallet/src/ui/react/ConnectAccount/ConnectWithEmoji.tsx

```diff
@@ -1,25 +1,35 @@
 import React from 'react';
 import {EmojiPanel} from '@universal-login/react';
-import {generateCode, TEST_ACCOUNT_ADDRESS} from '@universal-login/commons';
 import vault1x from './../../assets/illustrations/vault.png';
 import vault2x from './../../assets/illustrations/vault@2x.png';
+import {useServices, useRouter, useAsync} from '../../hooks';
 
-export const ConnectWithEmoji = () => (
-  <div className="main-bg">
-    <div className="box-wrapper">
-      <div className="box">
-        <div className="box-header">
-          <h1 className="box-title">Confirmation</h1>
-        </div>
-        <div className="box-content connect-emoji-content">
-          <div className="connect-emoji-section">
-            <img src={vault1x} srcSet={vault2x} alt="avatar" className="connect-emoji-img"/>
-            <p className="box-text connect-emoji-text">Thanks, now check another device controling this account and enter the emojis in this order:</p>
-            <EmojiPanel code={generateCode(TEST_ACCOUNT_ADDRESS)}/>
-            <button className="button-secondary connect-emoji-btn">Deny</button>
+interface ConnectWithEmojiProps {
+  name: string;
+}
+
+export const ConnectWithEmoji = ({name}: ConnectWithEmojiProps) => {
+  const {connectToWallet} = useServices();
+  const [connectValues] = useAsync(async () => connectToWallet(name, () =>  history.push('/')), []);
```

Comment:
extract:
`const navigateToMainScreen = () => history.push('/')` and pass as ConnectWithEmoji prop `onSuccess`.

Reply:
`const [{securityCode}, error] = `

Add error handling

Reply:
`connectValues.unsubscribe` is never used, shouldn't it be used?  @JustynaBroniszewska

Reply:
Yes, we should use `connectValues .unsubscribe` on the `deny` button click.

Reply:
1. We cannot extract () => history.push('/'). When we do it we breaking the rule 'Don’t call Hooks from regular JavaScript functions'
2. We need to have the possibility to really deny the request by sdk. Not just unsubscribe. Next PR?